pkg/operator/controller/controller_router_deployment.go

```diff
@@ -264,10 +267,17 @@ func (r *reconciler) updateRouterDeployment(current, desired *appsv1.Deployment)
 // deploymentConfigChanged checks if current config matches the expected config
 // for the cluster ingress deployment and if not returns the updated config.
 func deploymentConfigChanged(current, expected *appsv1.Deployment) (bool, *appsv1.Deployment) {
+	tolerationsEqual := true
+	if expected.Spec.Template.Spec.Tolerations != nil {
```

Comment:
Do we need this `if`? `cmp.Equal` should handle nil values correctly (and `cmpopts.EquateEmpty()` should make it treat nil and an empty map as equal).

If the `if` can be deleted, we can do it in a follow-up—it's a minor issue, and we need to get this PR merged.

Reply:
Let me test without the `cmp.Equal` wrapped in the `if` block. I was having problems with a unit test panicing and I thought this was the answer.